From 2017dfed93f63bec81b5d150bb9c7cff2de9b785 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 19 Jan 2022 05:51:59 +0200
Subject: [PATCH] [SW-58071] common: replace register usage with regmap

Since regmap and regmask files are added for gaudi2 and greco,
this patch replaces register name usage in common files with
regmap names.

Change-Id: I853c1720b2f4215ef4705c9008e1f5a753994190
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index d92e3c7..025b122 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -31,9 +31,9 @@
 {
 	uint32_t reg;
 
-	reg = regmap_readl(mmPSOC_SECURITY_SW_STATUS_0);
+	reg = regmap_readl(mmSW_STATUS_0);
 	reg |= BIT(progress);
-	regmap_writel(reg, mmPSOC_SECURITY_SW_STATUS_0);
+	regmap_writel(reg, mmSW_STATUS_0);
 }
 
 static void arcboot_rom_status_update_bringup(enum arcboot_rom_progress progress,
@@ -41,9 +41,9 @@
 {
 	uint32_t reg;
 
-	reg = regmap_readl(mmPSOC_SECURITY_SW_STATUS_0);
+	reg = regmap_readl(mmSW_STATUS_0);
 	reg |= BIT(progress + position);
-	regmap_writel(reg, mmPSOC_SECURITY_SW_STATUS_0);
+	regmap_writel(reg, mmSW_STATUS_0);
 }
 
 void arcboot_rom_status_update_location(uint8_t progress,
@@ -51,9 +51,9 @@
 {
 	uint32_t reg;
 
-	reg = regmap_readl(mmPSOC_SECURITY_SW_STATUS_0);
+	reg = regmap_readl(mmSW_STATUS_0);
 	reg |= loc << progress;
-	regmap_writel(reg, mmPSOC_SECURITY_SW_STATUS_0);
+	regmap_writel(reg, mmSW_STATUS_0);
 }
 
 static void arcboot_rom_set_efuse_read(void)
@@ -61,9 +61,9 @@
 	uint32_t reg;
 
 	/* validation succeeded, set only validation pass*/
-	reg = regmap_readl(mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2);
+	reg = regmap_readl(mmCOLD_RST_FLOPS_2);
 	reg |= COLD_RST_FLOPS_EFUSE_READ_MASK;
-	regmap_writel(reg, mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2);
+	regmap_writel(reg, mmCOLD_RST_FLOPS_2);
 }
 
 void arcboot_rom_init_sp_sram(void)
@@ -74,22 +74,22 @@
 
 	/* Use memblock to initialize scratchpad memory */
 	regmap_writel((uint32_t)HL_SP_SRAM_BASE,
-			mmPSOC_GLOBAL_CONF_MEM_CPY_DEST_ADDR_L);
+			mmMEM_CPY_DEST_ADDR_L);
 	regmap_writel((uint32_t)(HL_SP_SRAM_BASE >> 32),
-			mmPSOC_GLOBAL_CONF_MEM_CPY_DEST_ADDR_H);
+			mmMEM_CPY_DEST_ADDR_H);
 
-	mem_cfg = regmap_readl(mmPSOC_GLOBAL_CONF_MEM_CPY_CTRL2);
+	mem_cfg = regmap_readl(mmMEM_CPY_CTRL2);
 	mem_cfg &= ~(cfg_mask);
 	/* set size of 64k in granularity of 16 bytes */
 	mem_cfg |= (SP_SRAM_SIZE_16B_GRANULARITY & cfg_mask);
-	mem_cfg |= (0x1 << PSOC_GLOBAL_CONF_MEM_CPY_CTRL2_USE_CONST_SHIFT);
+	mem_cfg |= (0x1 << MEM_CPY_CTRL2_USE_CONST_SHIFT);
 
-	regmap_writel(mem_cfg, mmPSOC_GLOBAL_CONF_MEM_CPY_CTRL2);
+	regmap_writel(mem_cfg, mmMEM_CPY_CTRL2);
 
 	/* Enable copy and wait for completion */
-	regmap_writel(0x1, mmPSOC_GLOBAL_CONF_MEM_CPY_CTRL);
+	regmap_writel(0x1, mmMEM_CPY_CTRL);
 	while (1) {
-		cpy_sts = regmap_readl(mmPSOC_GLOBAL_CONF_MEM_CPY_STATUS);
+		cpy_sts = regmap_readl(mmMEM_CPY_STATUS);
 		if (cpy_sts)
 			break;
 		/*
@@ -137,13 +137,13 @@
 	spi_flash_set_byte_swap(DISABLE_BYTE_SWAP);
 
 	/* disable global masking of prstn reset */
-	regmap_writel(0, mmPSOC_GLOBAL_CONF_PRSTN_MASK);
+	regmap_writel(0, mmPRSTN_MASK);
 
 	/* on warm reset change default to reset ARC and PSOC */
 	update_warm_reset_config();
 
 	/* read from cold reset flops the recovery and validation flag */
-	reg = regmap_readl(mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2);
+	reg = regmap_readl(mmCOLD_RST_FLOPS_2);
 	arcboot_rom_global.recovery_flag =
 		(reg & COLD_RST_FLOPS_RECOVERY_FLAG_MASK) >>
 		 COLD_RST_FLOPS_RECOVERY_FLAG_SHIFT;
@@ -156,7 +156,7 @@
 		(reg & COLD_RST_FLOPS_EFUSE_READ_MASK) >>
 		 COLD_RST_FLOPS_EFUSE_READ_SHIFT;
 
-	regmap_writel(SPI_FLASH_BAUDR_DIVISOR, mmPSOC_QSPI_BAUDR);
+	regmap_writel(SPI_FLASH_BAUDR_DIVISOR, mmQSPI_BAUDR);
 
 	arcboot_rom_set_spi_write_limit();
 
@@ -173,7 +173,7 @@
 
 	if ((arcboot_rom_global.config_fuse &
 	     EFUSE_SYSCFG_GPIO02_RECOVERY_CONTROL) == 0)
-		override = regmap_readl(mmPSOC_GPIO0_DATA) & GPIO02_MASK;
+		override = regmap_readl(mmGPIO0_DATA) & GPIO02_MASK;
 
 	/* if recovery flag is set, bringup PCIe in recovery mode */
 	if (arcboot_rom_global.recovery_flag || (override == 0))
@@ -191,10 +191,10 @@
 	uint32_t reg;
 
 	/* validation failed, set recovery flag and clear validation */
-	reg = regmap_readl(mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2);
+	reg = regmap_readl(mmCOLD_RST_FLOPS_2);
 	reg |= COLD_RST_FLOPS_RECOVERY_FLAG_MASK;
 	reg &= ~COLD_RST_FLOPS_VALIDATION_PASS_MASK;
-	regmap_writel(reg, mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2);
+	regmap_writel(reg, mmCOLD_RST_FLOPS_2);
 }
 
 static void arcboot_rom_clear_recovery_flag(void)
@@ -202,9 +202,9 @@
 	uint32_t reg;
 
 	/* validation succeeded, set only validation pass*/
-	reg = regmap_readl(mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2);
+	reg = regmap_readl(mmCOLD_RST_FLOPS_2);
 	reg &= ~COLD_RST_FLOPS_RECOVERY_FLAG_MASK;
-	regmap_writel(reg, mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2);
+	regmap_writel(reg, mmCOLD_RST_FLOPS_2);
 }
 
 static void arcboot_rom_fuse(void)
diff --git a/common/dma.c b/common/dma.c
index a49eb94..98aad95 100644
--- a/common/dma.c
+++ b/common/dma.c
@@ -18,12 +18,10 @@
 	uint32_t spi_dma_dst_addr_h = DMA_ADDRESS_HIGH(dest_offset);
 	uint16_t flash_offset_1k = flash_offset / DMA_SRC_BLOCK_SIZE;
 	uint16_t image_size_4b = (image_size / DMA_DST_BLOCK_SIZE);
-	uint32_t fld_size_shift = get_fld_size_shift();
+	uint32_t fld_size_shift = SPI_DMA_CTRL_FLD_SIZE_SHIFT;
 
-	regmap_writel(spi_dma_dst_addr_l,
-		      mmPSOC_GLOBAL_CONF_SPI_DMA_DST_ADDR_L);
-	regmap_writel(spi_dma_dst_addr_h,
-		      mmPSOC_GLOBAL_CONF_SPI_DMA_DST_ADDR_H);
+	regmap_writel(spi_dma_dst_addr_l, mmSPI_DMA_DST_ADDR_L);
+	regmap_writel(spi_dma_dst_addr_h, mmSPI_DMA_DST_ADDR_H);
 
 	/* enable DMA transfer */
 	reg_spi_dma_ctrl_s = SPI_DMA_ENABLE;
@@ -36,9 +34,9 @@
 
 	/* spi offset in 1k chunks */
 	reg_spi_dma_ctrl_s |= (uint32_t)flash_offset_1k <<
-			      PSOC_GLOBAL_CONF_SPI_DMA_CTRL_ADDR_SHIFT;
+			      SPI_DMA_CTRL_ADDR_SHIFT;
 
-	regmap_writel(reg_spi_dma_ctrl_s, mmPSOC_GLOBAL_CONF_SPI_DMA_CTRL);
+	regmap_writel(reg_spi_dma_ctrl_s, mmSPI_DMA_CTRL);
 }
 
 void dma_copy_image(uint16_t image_size, uint32_t flash_offset,
@@ -56,12 +54,12 @@
 	int crc_error = 0;
 	do {
 		reg_spi_dma_status_s =
-		 regmap_readl(mmPSOC_GLOBAL_CONF_SPI_DMA_STATUS);
+		 regmap_readl(mmSPI_DMA_STATUS);
 	} while ((reg_spi_dma_status_s &
-		  PSOC_GLOBAL_CONF_SPI_DMA_STATUS_DONE_MASK) == 0);
+		  SPI_DMA_STATUS_DONE_MASK) == 0);
 	crc_error = (reg_spi_dma_status_s &
-		     PSOC_GLOBAL_CONF_SPI_DMA_STATUS_ERROR_MASK) >>
-		     PSOC_GLOBAL_CONF_SPI_DMA_STATUS_ERROR_SHIFT;
+		     SPI_DMA_STATUS_ERROR_MASK) >>
+		     SPI_DMA_STATUS_ERROR_SHIFT;
 
 	return crc_error;
 }
diff --git a/common/efuse.c b/common/efuse.c
index 59e30cf..5c5eb59 100644
--- a/common/efuse.c
+++ b/common/efuse.c
@@ -31,12 +31,12 @@
 	if (arcboot_rom_global.efuse_read == 1)
 		return -1;
 
-	read_reg = get_efuse_read_reg();
+	read_reg = mmEFUSE_READ_CTRL;
 
 	/* TODO: ask suyash to test time of this function */
 	/* TODO: facto 10 of function time ~10mS*/
 	regmap_writel(EFUSE_READ_RANGE_ALL,
-		      mmPSOC_EFUSE_READ_ADDRESS);
+		      mmEFUSE_READ_ADDRESS);
 
 	regmap_writel(1, read_reg);
 
@@ -66,12 +66,11 @@
 
 	/* read value from bank0 */
 	offset = (word * sizeof(uint32_t));
-	*val = regmap_readl(mmPSOC_EFUSE_DATA_0 + offset);
+	*val = regmap_readl(mmEFUSE_DATA_0 + offset);
 
 	/* read value from bank1 and OR with value from bank0 */
 	offset = ((FUSE_WORDS_PER_BANK + word) * sizeof(uint32_t));
-	*val |= regmap_readl(mmPSOC_EFUSE_DATA_0 + offset);
-
+	*val |= regmap_readl(mmEFUSE_DATA_0 + offset);
 	return 0;
 }
 
diff --git a/common/pcie.c b/common/pcie.c
index 50c8f4d..47ce067 100644
--- a/common/pcie.c
+++ b/common/pcie.c
@@ -18,8 +18,8 @@
 	uint8_t pcie_prstn;
 
 	do {
-		pcie_prstn = regmap_readl(mmPSOC_GLOBAL_CONF_PRSTN);
-		pcie_prstn &= PSOC_GLOBAL_CONF_PRSTN_MASK_IND_MASK;
+		pcie_prstn = regmap_readl(mmPRSTN);
+		pcie_prstn &= PRSTN_MASK_IND_MASK;
 	} while (pcie_prstn == assert);
 }
 
@@ -52,12 +52,12 @@
 
 			pcie_phy_set_mpll_a_b_ssc_enable();
 
-			regmap_writel(0x5000, mmPCIE_PHY_EXT_MPLLB_SSC_PEAK);
+			regmap_writel(0x5000, mmEXT_MPLLB_SSC_PEAK);
 			regmap_writel(0x19CE,
-				      mmPCIE_PHY_EXT_MPLLB_SSC_STEP_SIZE);
-			regmap_writel(0x6666, mmPCIE_PHY_EXT_MPLLA_SSC_PEAK);
+				      mmEXT_MPLLB_SSC_STEP_SIZE);
+			regmap_writel(0x6666, mmEXT_MPLLA_SSC_PEAK);
 			regmap_writel(0x2108,
-				      mmPCIE_PHY_EXT_MPLLA_SSC_STEP_SIZE);
+				      mmEXT_MPLLA_SSC_STEP_SIZE);
 		}
 
 		pcie_phy_set_ext_tx_rx_ctrl();
@@ -66,7 +66,7 @@
 
 		pcie_phy_set_protocol0_ext_tx_rx();
 
-		regmap_writel(1, mmPCIE_PHY_PHY_EXT_CTRL_SEL);
+		regmap_writel(1, mmPHY_EXT_CTRL_SEL);
 	}
 }
 
@@ -76,24 +76,24 @@
 	uint8_t ltssm_en;
 
 	pcie_loop_while_prstn(PCIE_PRSTN_ASSERT);
-	ltssm_en = regmap_readl(mmPCIE_AUX_LTSSM_EN) &
-				PCIE_AUX_LTSSM_EN_IND_MASK;
+	ltssm_en = regmap_readl(mmLTSSM_EN) &
+				LTSSM_EN_IND_MASK;
 
 	if (!ltssm_en) {
-		regmap_writel(0, mmPSOC_GLOBAL_CONF_PCIE_PHY_RST_N);
+		regmap_writel(0, mmPCIE_PHY_RST_N);
 
 		pcie_phy_para_sel_set();
 
 		/* gpio bootstrap pin gpio17 = 0 */
-		boot_strap = regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L);
+		boot_strap = regmap_readl(mmBOOT_STRAP_PINS_L);
 
 		if (get_sris_mode_from_strap()) {
-			regmap_writel(1, mmPCIE_PHY_PIPE_RX_SRIS_MODE_EN);
-			regmap_writel(1, mmPCIE_AUX_APP_SRIS_MODE);
+			regmap_writel(1, mmPIPE_RX_SRIS_MODE_EN);
+			regmap_writel(1, mmAPP_SRIS_MODE);
 		}
 
 		pcie_wa_snps_case_01038891(boot_strap);
-		regmap_writel(1, mmPSOC_GLOBAL_CONF_PCIE_PHY_RST_N);
+		regmap_writel(1, mmPCIE_PHY_RST_N);
 
 		pcie_wait_for_sram_init_done();
 
@@ -101,15 +101,18 @@
 
 		if ((arcboot_rom_global.config_fuse & EFUSE_SYSCFG_PCIE_GEN_LIMIT) == 0) {
 			/* limit PCIe link gen to 1 */
-			link_ctrl = regmap_readl(mmPCIE_DBI_LINK_CAPABILITIES_REG);
-			link_ctrl &= ~PCIE_DBI_LINK_CAPABILITIES_REG_PCIE_CAP_MAX_LINK_SPEED_MASK;
-			link_ctrl |= (1 << PCIE_DBI_LINK_CAPABILITIES_REG_PCIE_CAP_MAX_LINK_SPEED_SHIFT);
-			regmap_writel(link_ctrl, mmPCIE_DBI_LINK_CAPABILITIES_REG);
+			link_ctrl = regmap_readl(mmLINK_CAPABILITIES_REG);
+			link_ctrl &=
+				~LINK_CAPABILITIES_REG_PCIE_CAP_MAX_LINK_SPEED_MASK;
+			link_ctrl |= (1 << LINK_CAPABILITIES_REG_PCIE_CAP_MAX_LINK_SPEED_SHIFT);
+			regmap_writel(link_ctrl, mmLINK_CAPABILITIES_REG);
 
-			link_ctrl = regmap_readl(mmPCIE_DBI_LINK_CONTROL2_LINK_STATUS2_REG);
-			link_ctrl &= ~PCIE_DBI_LINK_CONTROL2_LINK_STATUS2_REG_PCIE_CAP_TARGET_LINK_SPEED_MASK;
-			link_ctrl |= (1 << PCIE_DBI_LINK_CONTROL2_LINK_STATUS2_REG_PCIE_CAP_TARGET_LINK_SPEED_SHIFT);
-			regmap_writel(link_ctrl, mmPCIE_DBI_LINK_CONTROL2_LINK_STATUS2_REG);
+			link_ctrl = regmap_readl(mmLINK_CONTROL2_LINK_STATUS2_REG);
+			link_ctrl &=
+				~LINK_CONTROL2_LINK_STATUS2_REG_PCIE_CAP_TARGET_LINK_SPEED_MASK;
+			link_ctrl |= (1 <<
+				LINK_CONTROL2_LINK_STATUS2_REG_PCIE_CAP_TARGET_LINK_SPEED_SHIFT);
+			regmap_writel(link_ctrl, mmLINK_CONTROL2_LINK_STATUS2_REG);
 		}
 
 		/* Configure IATU */
@@ -119,12 +122,11 @@
 		enable_flr_handling_fsm();
 
 		/* disable HOT reset handling by HW state machine */
-		regmap_writel(PCIE_WRAP_CPU_HOT_RST_CTRL_MASK,
-			      mmPCIE_WRAP_CPU_HOT_RST);
+		regmap_writel(CPU_HOT_RST_CTRL_MASK,
+			      mmCPU_HOT_RST);
 
-		regmap_writel(0x3, mmPCIE_AUX_LTSSM_EN);
+		regmap_writel(0x3, mmLTSSM_EN);
 	}
-
 }
 
 void pcie_recovery(void)
@@ -142,7 +144,7 @@
 	check_and_enable_4b_termination();
 
 	/* Disable PCIe error responses (both HBW and LBW) */
-	regmap_writel(0x1, mmPCIE_WRAP_PCIE_RSP_OK);
+	regmap_writel(0x1, mmPCIE_RSP_OK);
 
 	security_debug_if_secure();
 	security_debug_if_shared();
diff --git a/common/security.c b/common/security.c
index eeb46c8..2fca9a5 100644
--- a/common/security.c
+++ b/common/security.c
@@ -15,23 +15,23 @@
 {
 	/* Configure SMI,I2C and JTAG to be "secure" */
 	regmap_writel(MSTR_IF_CORE_HBW_AXPROT_OVRD_OVRD_EN,
-		      mmI2C_S_MSTR_IF_CORE_HBW_AXPROT_OVRD);
+		      mmI2C_S_MSTR_IF_CORE_HBW_AXPROT_OVD);
 	regmap_writel(MSTR_IF_CORE_HBW_AXPROT_OVRD_OVRD_EN,
-		      mmSMI_MSTR_IF_CORE_HBW_AXPROT_OVRD);
+		      mmSMI_MSTR_IF_CORE_HBW_AXPROT_OVD);
 	regmap_writel(MSTR_IF_CORE_HBW_AXPROT_OVRD_OVRD_EN,
-		      mmJT_MSTR_IF_CORE_HBW_AXPROT_OVRD);
+		      mmJT_MSTR_IF_CORE_HBW_AXPROT_OVD);
 }
 
 void security_spi_dma_mmu_bypass(void)
 {
 	regmap_writel(MSTR_IF_AXUSER_HB_SEC_MMBP,
-		      mmPSOC_GLOBAL_CONF_SPI_DMA_AWUSER);
+		      mmSPI_DMA_AWUSER);
 }
 
 void security_mem_cpy_mmu_bypass(void)
 {
 	regmap_writel(MSTR_IF_AXUSER_HB_SEC_MMBP,
-		      mmPSOC_GLOBAL_CONF_MEM_CPY_AXUSER);
+		      mmMEM_CPY_AXUSER);
 }
 
 /* If DBG I/F were not disabled, program the RR to allow access
@@ -52,17 +52,17 @@
 void security_i2c_s_enable(void)
 {
 	/* Slave address */
-	regmap_writel(0x6A, mmPSOC_I2C_S_IC_SAR);
+	regmap_writel(0x6A, mmI2C_S_IC_SAR);
 	/* IC_RESTART_EN and speed: standard mode - 100 kbit/s */
-	regmap_writel(0x22, mmPSOC_I2C_S_IC_CON);
+	regmap_writel(0x22, mmI2C_S_IC_CON);
 	/* Enable all interrupt via mask */
-	regmap_writel(0x7FFF, mmPSOC_I2C_S_IC_INTR_MASK);
+	regmap_writel(0x7FFF, mmI2C_S_IC_INTR_MASK);
 	/* I2C Receive FIFO Threshold to 0 */
-	regmap_writel(0x0, mmPSOC_I2C_S_IC_RX_TL);
+	regmap_writel(0x0, mmI2C_S_IC_RX_TL);
 	/* I2C Transmit FIFO Threshold to 7 */
-	regmap_writel(0x7, mmPSOC_I2C_S_IC_TX_TL);
+	regmap_writel(0x7, mmI2C_S_IC_TX_TL);
 	/* Enable I2C slave controller */
-	regmap_writel(0x1, mmPSOC_I2C_S_IC_ENABLE);
+	regmap_writel(0x1, mmI2C_S_IC_ENABLE);
 }
 
 /* minimal allowed security version is defined by a 128bit map, to find this
@@ -181,7 +181,7 @@
 	key_revoke &= EFUSE_SYSCFG_REVOKE_KEY_MASK;
 
 	/* store key_revoke map in PSOC */
-	regmap_writel(key_revoke, mmPSOC_SECURITY_KEY_REVOKE);
+	regmap_writel(key_revoke, mmSECURITY_KEY_REVOKE);
 
 	/* store key hashes in Scratch PAD Sram */
 	sp_sram_writel(key_revoke,
@@ -216,9 +216,10 @@
 		fw_version_sec = efuse_otp_get_fw_version_security();
 		break;
 	}
+
 	/* store minimum allowed security version in PSOC */
 	regmap_writel(SECURITY_FW_VERSION_FROM_SECURITY_BYTE(fw_version_sec),
-		      mmPSOC_SECURITY_MIN_FW_VERSION);
+		      mmSECURITY_MIN_FW_VERSION);
 
 	/* store minimum allowed security version in Scratch PAD Sram */
 	sp_sram_writel(SECURITY_FW_VERSION_FROM_SECURITY_BYTE(fw_version_sec),
@@ -230,9 +231,9 @@
 	uint32_t offset;
 
 	if (index == ARCBOOT_SECONDARY)
-		offset = mmPSOC_SECURITY_SEC_IMG_FW_VER;
+		offset = mmSECURITY_SEC_IMG_FW_VER;
 	else
-		offset = mmPSOC_SECURITY_PRI_IMG_FW_VER;
+		offset = mmSECURITY_PRI_IMG_FW_VER;
 
 	regmap_writel(fw_ver, offset);
 }
@@ -243,21 +244,20 @@
 	uint32_t reg, offset;
 
 	if (index == ARCBOOT_SECONDARY)
-		offset = mmPSOC_SECURITY_SEC_IMG_STATUS;
+		offset = mmSECURITY_SEC_IMG_STATUS;
 	else
-		offset = mmPSOC_SECURITY_PRI_IMG_STATUS;
+		offset = mmSECURITY_PRI_IMG_STATUS;
 
 	reg = regmap_readl(offset);
 	switch (img_stat) {
 	case IMG_STATUS_FW_VERSION_LOW:
-		reg |= 1 << PSOC_SECURITY_PRI_IMG_STATUS_FW_VER_LOWER_SHIFT;
+		reg |= 1 << SECURITY_PRI_IMG_STATUS_FW_VER_LOWER_SHIFT;
 		break;
 	case IMG_STATUS_SIG_AUTH_FAIL:
-		reg |= 1 <<
-		       PSOC_SECURITY_PRI_IMG_STATUS_AUTHENTICATE_FAIL_SHIFT;
+		reg |= 1 << SECURITY_PRI_IMG_STATUS_AUTHENTICATE_FAIL_SHIFT;
 		break;
 	case IMG_STATUS_CRC_FAIL:
-		reg |= 1 << PSOC_SECURITY_PRI_IMG_STATUS_CRC_FAIL_SHIFT;
+		reg |= 1 << SECURITY_PRI_IMG_STATUS_CRC_FAIL_SHIFT;
 		break;
 	default:
 		reg = 0;
@@ -271,36 +271,35 @@
 {
 	uint32_t reg;
 
-	reg = regmap_readl(mmPSOC_SECURITY_STATUS);
+	reg = regmap_readl(mmSECURITY_STATUS);
 	switch (sec_stat) {
 	case STATUS_IMG_VALID:
-		reg |= 1 << PSOC_SECURITY_STATUS_IMG_VALIDATED_SHIFT;
+		reg |= 1 << SECURITY_STATUS_IMG_VALIDATED_SHIFT;
 		break;
 	case STATUS_FW_UPGRADE:
-		reg |= 1 << PSOC_SECURITY_STATUS_FW_UPGRADE_SHIFT;
+		reg |= 1 << SECURITY_STATUS_FW_UPGRADE_SHIFT;
 		break;
 	case STATUS_KEY_REVOKED:
-		reg |= 1 << PSOC_SECURITY_STATUS_KEY_REVOKED_SHIFT;
+		reg |= 1 << SECURITY_STATUS_KEY_REVOKED_SHIFT;
 		break;
 	case STATUS_KEY_REVOKED_INDX:
-		reg |= 1 << PSOC_SECURITY_STATUS_KEY_REVOKED_IND_SHIFT;
+		reg |= 1 << SECURITY_STATUS_KEY_REVOKED_IND_SHIFT;
 		break;
 	case STATUS_MIN_FW_MODIFIED:
-		reg |= 1 << PSOC_SECURITY_STATUS_MIN_FW_MODIFIED_SHIFT;
+		reg |= 1 << SECURITY_STATUS_MIN_FW_MODIFIED_SHIFT;
 		break;
 	case STATUS_FW_VER_INC:
-		reg |= 1 << PSOC_SECURITY_STATUS_FW_VER_INC_SHIFT;
+		reg |= 1 << SECURITY_STATUS_FW_VER_INC_SHIFT;
 		break;
 	case STATUS_SPI_IMG_MODIFIES:
-		reg |= index << PSOC_SECURITY_STATUS_SPI_IMG_MODIFIES_SHIFT;
+		reg |= index << SECURITY_STATUS_SPI_IMG_MODIFIES_SHIFT;
 		break;
 	case STATUS_FW_AUTH_FAIL:
-		reg |= 1 << PSOC_SECURITY_STATUS_FW_AUTHENTICAION_FAIL_SHIFT;
+		reg |= 1 << SECURITY_STATUS_FW_AUTHENTICAION_FAIL_SHIFT;
 		break;
 	default:
 		reg = 0;
 	}
 
-	regmap_writel(reg, mmPSOC_SECURITY_STATUS);
-
+	regmap_writel(reg, mmSECURITY_STATUS);
 }
diff --git a/common/spi_flash.c b/common/spi_flash.c
index 376f627..4c3d3cf 100644
--- a/common/spi_flash.c
+++ b/common/spi_flash.c
@@ -16,16 +16,16 @@
 
 	/* clear BYTE SWAP in SPI DMA so we dont switch endianness */
 	reg_spi_direct_wr_rd_ctrl =
-		regmap_readl(mmPSOC_GLOBAL_CONF_SPI_DIRECT_WR_RD_CTRL);
+		regmap_readl(mmSPI_DIRECT_WR_RD_CTRL);
 	if (enable)
 		reg_spi_direct_wr_rd_ctrl |=
-			PSOC_GLOBAL_CONF_SPI_DIRECT_WR_RD_CTRL_BYTE_SWAP_MASK;
+			SPI_DIRECT_WR_RD_CTRL_BYTE_SWAP_MASK;
 	else
 		reg_spi_direct_wr_rd_ctrl &=
-			~PSOC_GLOBAL_CONF_SPI_DIRECT_WR_RD_CTRL_BYTE_SWAP_MASK;
+			~SPI_DIRECT_WR_RD_CTRL_BYTE_SWAP_MASK;
 
 	regmap_writel(reg_spi_direct_wr_rd_ctrl,
-		      mmPSOC_GLOBAL_CONF_SPI_DIRECT_WR_RD_CTRL);
+		      mmSPI_DIRECT_WR_RD_CTRL);
 }
 
 uint32_t spi_flash_readl(uint32_t offset)
@@ -106,11 +106,11 @@
 void spi_flash_otp_access(uint8_t enable)
 {
 	if (enable)
-		regmap_writel(FLASH_OTP_CMD_ENTER, mmPSOC_QSPI_DR0);
+		regmap_writel(FLASH_OTP_CMD_ENTER, mmQSPI_DR0);
 	else
-		regmap_writel(FLASH_OTP_CMD_EXIT, mmPSOC_QSPI_DR0);
+		regmap_writel(FLASH_OTP_CMD_EXIT, mmQSPI_DR0);
 
-	regmap_writel(FLASH_OTP_START_TRANSFER, mmPSOC_QSPI_SER);
+	regmap_writel(FLASH_OTP_START_TRANSFER, mmQSPI_SER);
 }
 
 void spi_flash_otp_get_revoke(uint32_t *key_revoke)
diff --git a/include/arcboot.h b/include/arcboot.h
index de42345..c3a25c2 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -11,6 +11,8 @@
 #include <stdio.h>
 #include <stdint.h>
 #include "soc.h"
+#include "regmap.h"
+#include "regmask.h"
 
 #define BIT(nr)					(1 << (nr))
 #define ARRAY_SIZE(x)				(sizeof(x) / sizeof((x)[0]))
